.gitignore

```diff
@@ -1,2 +1,5 @@
 .vscode
 generated_docs
+build
+install
```

Comment:
what is this?

Reply:
It guards against accidentally committing the build/install directories. Since I was building/testing locally, I added it in here, but I can move it to a different PR if that's preferred.

Reply:
please move this in a separate PR

Reply:
Removed from this PR